dot/state/grandpa.go

```diff
@@ -73,7 +73,7 @@ func NewGrandpaStateFromGenesis(db chaindb.Database, bs *BlockState,
 }
 
 // NewGrandpaState returns a new GrandpaState
-func NewGrandpaState(db chaindb.Database, bs *BlockState) *GrandpaState {
+func NewGrandpaState(db *chaindb.BadgerDB, bs *BlockState) *GrandpaState {
```

Comment:
This is left as an implementation on purpose, since it should be changed by issue #2981